examples/X_NUCLEO_IKS02A1_Audio_Button/X_NUCLEO_IKS02A1_Audio_Button.ino

```diff
@@ -0,0 +1,233 @@
+      }
+    
+      if (!(myFile = SD.open(file_name, FILE_WRITE))) {
+        Serial.println("Failed to Inizialize File\n");
+        while(1);
+      }
+    
+      WAV.header_init(pHeader, 16000, 16);
+    
+      myFile.write(pHeader, 44);
+      myFile.close();
+      Serial.flush();
+
+      start_stop = true;
+      Serial.println("RECORDING");
+      PDM.Record(sampleBuffer);
+      
+    }else{
+      start_stop = false;
+      if (PDM.Stop() == PDM_ERROR)
+      {
+        Serial.println("PDM_ERROR");
+      }
+      Serial.println("STOP");
+      
+      /* Update and Writing header */
+      if (!(myFile = SD.open(file_name, O_WRITE))) {
+        Serial.println("Failed to Open File\n");
+        while(1);
+      }
+    
+      WAV.header_update(pHeader, &byteswritten);
+      if (!myFile.seek(0)) {
+        Serial.println("Failed to Update Header\n");
+        while(1);
+      }
+      myFile.write(pHeader, 44);
+      myFile.close();   
+
+    }
+    button = false;
+  }
+}
+
+
+extern "C" void foo()
+{
+  digitalWrite(A0, !digitalRead(A0));
+  digitalWrite(A1, !digitalRead(A1));
+  //testing_writing();
+  
+  /*pdmtopcm*/
+  //swap_reverse32(sampleBuffer, sizeof(sampleBuffer));
+
+  for (int i = 0; i < N_MS_PER_INTERRUPT; i++)
+  {
+    pdm2pcm((uint8_t *) & (sampleBuffer[i * ((BLOCK_SIZE/8)/2)]), &pcmSamples[i * PCM_BUFLEN], BLOCK_SIZE);
+  }
+
+  /* Writing on SD */
+  myFile = SD.open(file_name, O_WRITE);
+  myFile.seek(byteswritten);
+  for (int i = 0; i < PCM_BUFLEN * N_MS_PER_INTERRUPT; i++)
+  {
+    uint16_t data = (uint16_t)pcmSamples[i];
+
+    myFile.write((uint8_t)data);
+    myFile.write((uint8_t)(data>>8));
+    
+  }
+  
+  byteswritten += PCM_BUFLEN * N_MS_PER_INTERRUPT * sizeof(int16_t);
+  myFile.flush();
+  myFile.close();  
+
+  digitalWrite(A1, !digitalRead(A1));
+}
+
+void play_stop()
+{
+  button = true;
+}
+
+void swap(uint16_t* data, int data_length)
+{
+  uint8_t* data_8bit = (uint8_t *) data;
+  for(int i=0;i<(data_length/2);i++)
+  {
+    uint8_t temp = data_8bit[2*i];
+    data_8bit[2*i] = data_8bit[2*i+1];
+    data_8bit[2*i+1] = temp;   
+  }
+}
+
+uint8_t reverse(uint8_t b) {
+   b = (b & 0xF0) >> 4 | (b & 0x0F) << 4;
+   b = (b & 0xCC) >> 2 | (b & 0x33) << 2;
+   b = (b & 0xAA) >> 1 | (b & 0x55) << 1;
+   return b;
+}
+
+void swap_reverse(uint16_t* data, int data_length)
+{
+  uint8_t* data_8bit = (uint8_t *) data;
+  for(int i=0;i<(data_length/2);i++)
+  {
+    uint8_t temp = reverse(data_8bit[2*i]);
+    data_8bit[2*i] = reverse(data_8bit[2*i+1]);
+    data_8bit[2*i+1] = temp;   
+  }
+}
+
+void swap_reverse32(uint16_t* data, int data_length)
+{
+
+  uint32_t *data_32bit = (uint32_t *) data;
+  for(int i=0;i<(data_length/4);i++)
+  {
+    data_32bit[i] = __ROR(__RBIT(data_32bit[i]), 16);  
+  }
+}
+
+void testing_writing()
+{
+   /* Writing on SD */
+  myFile = SD.open("myTest.wav", O_WRITE);
+  
+  for(int i = 0; i < 256;i++)
+    pcmSamples[i] = i;
```

Comment:
As a best practice always enclosed single line of code for loop/condition ... This will avoid any further issue if the code is updated
```suggestion
  for(int i = 0; i < 256;i++)
  {
    pcmSamples[i] = i;
  }
```